compiler/rustc_feature/src/builtin_attrs.rs

```diff
@@ -749,6 +749,12 @@ pub const BUILTIN_ATTRIBUTES: &[BuiltinAttribute] = &[
         template!(NameValueStr: "transparent|semitransparent|opaque"), ErrorFollowing,
         EncodeCrossCrate::Yes, "used internally for testing macro hygiene",
     ),
+    // TODO(ZuseZ4): What does cross crate actually mean here?
```

Comment:
If this attribute needs to be read on a foreign item, it needs to be set. But it can *not* be set if we encode this data somehow else (e.g. in another query result) -- for example, we don't encode `#[repr(transparent)]` cross-crate because we store the repr in `AdtDef`, which is encoded cross-crate.

Reply:
Autodiff can only get applied to function definitions, which I think are never a foreign item.
At least we can't apply it to extern "C" functions yet (although that would be extremely cool to support in the future), so I think it will always be local. Also, we did add a new Query related to Autodiff data, so I think it must be No.

Reply:
I tested it using `No` and unfortunately this error still persists.
I'm fine with merging regardless since it's better than nothing, but this is a bug that would be very good to fix:
https://github.com/g-bauer/enzyme_playground/tree/main?tab=readme-ov-file#problems
(In summary, if autodiff is applied in a dependency, we will execute the placeholder I generate here, and not the llvm-ir I add to the placeholder at rustc_codegen_llvm level).

Reply:
I feel like this needs to be `EncodeCrossCrate::Yes`, because AFAICT it's not encoded elsewhere?